src/sentry/workflow_engine/models/data_condition.py

```diff
@@ -86,44 +80,31 @@ def get_condition_result(self) -> DataConditionResult:
 
         return None
 
-    def get_condition_handler(self) -> DataConditionHandler[T] | None:
+    def evaluate_value(self, value: T) -> DataConditionResult:
         try:
             condition_type = Condition(self.type)
         except ValueError:
-            # If the type isn't in the condition, then it won't be in the registry either.
-            raise NoRegistrationExistsError(f"No registration exists for {self.type}")
-
-        return condition_handler_registry.get(condition_type)
+            logger.exception(
+                "Invalid condition type",
+                extra={"type": self.type, "id": self.id},
+            )
+            return None
 
-    def evaluate_value(self, value: T) -> DataConditionResult:
-        condition_handler: DataConditionHandler[T] | None = None
-        op: Callable | None = None
+        if condition_type in condition_ops:
+            # If the condition is a base type, handle it directly
+            op = condition_ops[Condition(self.type)]
+            result = op(cast(Any, value), self.comparison)
+            return self.get_condition_result() if result else None
```

Comment:
I think we should consider not having exceptions, and just have handlers for all types. Not a blocker for merging though